pageserver/src/tenant.rs

```diff
@@ -1859,11 +1891,7 @@ impl Tenant {
     #[allow(clippy::too_many_arguments)]
     pub(crate) async fn create_timeline(
```

Comment:
Just a thought, not suggesting you do it now. Could we lift the timeline creation stuff to a separate `impl Tenant` in a different module? Like we do in `pgdatadir_mapping.rs`.